exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -298,6 +299,11 @@ def encode(span_data) # rubocop:disable Metrics/MethodLength, Metrics/Cyclomatic
         rescue StandardError => e
           OpenTelemetry.handle_error(exception: e, message: 'unexpected error in OTLP::Exporter#encode')
           nil
+        ensure
+          stop = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+          duration_ms = 1000.0 * (stop - start)
+          @metrics_reporter.record_value('otel.otlp_exporter.encode_duration',
```

Comment:
If we do this one more time, it's worth a refactor. I am not super keen on the block-based approach if all our callsites are full methods. I think I'd prefer:
```ruby
def foo
  probe = start_probe('otel.otlp_exporter.foo')
  ...
ensure
  probe.finish
end
```
over:
```ruby
def foo
  @metrics_reporter.record_duration('otel.otlp_exporter.foo') do
    ...
  end
end
```
simply because it avoids deeper indentation as we add instrumentation to existing methods. The latter is better for measurements of small chunks of code within a larger method.